app/controllers/first_registration/prospective_students_controller.rb

```diff
@@ -41,5 +60,17 @@ def set_prospective_student
     def can_register?
       redirect_to(index_path, alert: t('.can_not_register')) unless @prospective_student.can_temporarily_register?
     end
+
+    def prospective_student_params
```

Comment:
Burası çok kalabalık görünüyor. Aklıma gelen ilk çözüm önerisini yazıyorum.


`concern/permitable.rb`
```ruby
# frozen_string_literal: true

module Permitable
  extend ActiveSupport::Concern

  EXCLUDED_ATTRIBUTES = %i[
    id created_at updated_at
  ].freeze

  class_methods do
    def permited_attributes(included: nil, excluded: nil)
      attribute_names.map(&:to_sym) + [*included] - (EXCLUDED_ATTRIBUTES + [*excluded])
    end
  end
end
```

```ruby
  params.require(:prospective_student).permit(*ProspectiveStudent.permited_attributes)
```

Reply:
@isubas `(EXCLUDED_ATTRIBUTES + [*excluded])` yerine `[*EXCLUDED_ATTRIBUTES, *excluded]` (splat yaparken concat'tan kaçınmaya çalış)

Reply:
Buna çeşitli sebeplerle katılmıyorum:

1) Estetik kaygılar varsayılan pratiklerin önüne geçmemeli. Modelin attribute sayısı çok diye bir şeyi concern'leştirmeyi anlamsız buluyorum. Concern kiri pisliği saklamak için kullandığımız bir yapı olmamalı. Concern özünde bir module ve burada module altında gruplayacak ortak bir sınıf davranışı görmüyorum.

2) Tüm geliştiricilerin öğrenmesi gereken yeni bir karmaşıklık ekleniyor, hem de yalnızca bir modelin attribute sayısı çok, gözüme hoş gelmedi diye.

3) En önemli madde. Evet, `included` ve `excluded` argümanlarına attribute verebiliyoruz ancak çok önemli bir şeyi gözden kaçırıyorsun, bu modele eklenecek herhangi bir kritik attribute bundan sonra **mutlaka** `excluded` olarak `permitted_attributes`'a gönderilmeli.

Yani Rails'in default whitelisting mekanizmasını bir anda blacklisting'e çeviriyorsun. Güvenlik açısından whitelisting blacklisting farklarına girmeyeceğim. Özetle şunu söyleyeyim, Rails varsayılan olarak gereken tüm parametreleri bize whitelist ettirirken salakça bir şey yapmıyor. Bu parametrelerin sayıca epey çok olabileceğini öngörememiş de değiller. Burada default davranışın whitelist'e force etmesinin tek bir sebebi var o da güvenlik.